lib/intercom/request.rb

```diff
@@ -180,6 +185,8 @@ def raise_application_errors_on_failure(error_list_details, http_code)
         raise Intercom::MultipleMatchingUsersError.new(error_details['message'], error_context)
       when 'resource_conflict'
         raise Intercom::ResourceNotUniqueError.new(error_details['message'], error_context)
+      when 'intercom_version_invalid'
+        raise Intercom::ApiVersionInvalid.new(error_details['message'], error_context)
```

Comment:
It would be good to add a test for this block

Reply:
Added in db3a03a